### PR TITLE
Support rootURL.

### LINKED ISSUE #29: Ember Asset loader does not take rootURL into consideration
Assets are loaded absolutely instead of relative to account for a rootURL setting

### PR DESCRIPTION
**Pull request by nathanhammond**
Supersedes #32. Closes #29.

### CODE REVIEW

File: lib/asset-manifest-generator.js [hunk 1]
```diff
@@ -100,9 +105,12 @@ AssetManifestGenerator.prototype.build = function() {
     else if (assetName && bundleName) {
       var assetType = assetName.split('.').pop();
 
+      // `entry` has no leading `/`.
+      // `prepend` should be normalized to have a leading and trailing `/`.
```
Comment (trentmwillis) [inline lib/asset-manifest-generator.js]: I feel like these two line comments should be moved inside the `if` below.

File: lib/manifest-generator.js [hunk 2]
```diff
@@ -19,8 +20,8 @@ var ManifestGenerator = Addon.extend({
    */
   contentFor: function(type, config) {
     var app = findHost(this);
-    var options = app && app.options && app.options.assetLoader;
-    if (type === 'head-footer' && !(options && options.noManifest)) {
+    var appBuildtimeOptions = app && app.options && app.options.assetLoader;
```
Comment (trentmwillis) [inline lib/manifest-generator.js]: Can we name this `assetLoaderOptions`, like we have it below?

File: lib/manifest-generator.js [hunk 3]
```diff
@@ -33,23 +34,70 @@ var ManifestGenerator = Addon.extend({
...
 
     if (assetLoaderOptions.noManifest) {
       return tree;
     }
 
+    // root's config/environment.js
+    var runtimeOptions = app.project.config(app.env);
+
+    // Normalize the `rootURL` per Ember CLI's pattern.
+    // Also checks `baseURL` to support the old `baseURL` setting.
```
Comment (trentmwillis) [inline lib/manifest-generator.js]: Is this really worth supporting? It's old, this addon is new, I'd prefer not supporting it.

**Review by trentmwillis — CHANGES_REQUESTED** [review]
Seems good on first pass. We should try to breakdown the `postprocessTree` function if possible.

File: lib/manifest-generator.js (hunk 3 above)
Comment (nathanhammond) [reply lib/manifest-generator.js]: In Ember CLI 2.6 we used `baseURL` and 2.7 we added `rootURL` and deprecated (but still support) `baseURL`-only setups. It's not old enough in my opinion to drop the branching support.

File: lib/asset-manifest-generator.js (hunk 1 above)
Comment (nathanhammond) [reply lib/asset-manifest-generator.js]: Expanded upon them for future travelers.

File: lib/manifest-generator.js (hunk 2 above)
Comment (nathanhammond) [reply lib/manifest-generator.js]: ☑️

**Comment by trentmwillis** [comment]
I think we could probably move the logic for constructing the `manifestOptions` into a separate function. I think that will fix the code climate check and keep the complexity low.

**Comment by trentmwillis** [comment]
Closing this as I believe we wound up not wanting to take this approach (and it has gone stale).

**Pull request closed without merging**